infra/base-images/base-builder/compile

```diff
@@ -40,12 +37,18 @@ then
     # break non MSan compiled programs.
     (cd "$MSAN_LIBS_PATH" && find . -name '*.a' -exec cp --parents '{}' / ';')
   fi
+
+  FUZZING_ENGINE_SANITIZER_FLAGS="$SANITIZER_FLAGS"
+fi
+
+if [[ $FUZZING_ENGINE != "none" ]]; then
+  # compile script might override environment, use . to call it.
+  SANITIZER_FLAGS="$FUZZING_ENGINE_SANITIZER_FLAGS" . compile_${FUZZING_ENGINE}
```

Comment:
i don't think any compile_* scripts currently do this, but would overriding SANITIZER_FLAGS in compile_* still work? I'm seeing the following:

```bash
$ cat test.sh
#!/bin/bash
export BLAH=123
$ BLAH=456 . ./test.sh
$ echo $BLAH
# nothing gets printed.
```

Reply:
actually, how about just separating SANITIZER_FLAGS from FUZZING_ENGINE_SANITIZER_FLAGS? i.e. make `compile_libfuzzer` (the only compile script which reads $SANITIZER_FLAGS) read $FUZZING_ENGINE_SANITIZER_FLAGS instead.

Reply:
Sounds good, done!